### DISCUSSION TITLE
How to reference s3 in `apiVersion: helm.toolkit.fluxcd.io/v2beta1`?

### DISCUSSION THREAD

**Discussion by hunkeelin** [fluxcd/flux2#3206] [General]
In flux v1 I can reference s3 by doing the following:
```
  1 ---
  2 apiVersion: helm.fluxcd.io/v1
  3 kind: HelmRelease
  4 metadata:
  5   name: config
  6   namespace: foo-apps
  7   labels:
  8     app: "config" # For label selectors (kustomize)
  9     dontpatch: me
 10   annotations:
 11     fluxcd.io/automated: "false"
 12 spec:
 13   releaseName: config
 14   chart:
 15     name: foo-apps
 16     repository: s3://helm-repo/
 17     version: 1.2.3
```

However I am having a hard time trying to compose the correct yaml for `apiVersion: helm.toolkit.fluxcd.io/v2beta1` The [documentation](https://fluxcd.io/flux/components/helm/api/) is confusing. 
```
  2 ---
  3 apiVersion: helm.toolkit.fluxcd.io/v2beta1
  4 kind: HelmRelease
  5 metadata:
  6   name: config
  7   namespace: flux-system
  8 spec:
  9   chart:
 10     spec:
 11       chart: foo-apps
 12       sourceRef:
 13         kind: bucket
 14         name: s3://helm-repo/
 15       version: 1.2.3
```

The above does not work

**Comment by hiddeco** [comment]
The way this is achieved is by creating a [`Bucket` source which refers to the S3 bucket](https://fluxcd.io/flux/components/source/buckets/), this can then be used as a source in the [HelmChart template](https://fluxcd.io/flux/components/helm/helmreleases/#helm-chart-template).
